pkg/catalogsourceconfig/configuring.go

```diff
@@ -62,13 +63,28 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *v2.CatalogSou
 	// in case of a failure.
 	r.cache.Set(out)
 
+	// If a source is not specified, check if there is an OperatorSource
+	// that contains all packages.
+	if out.Spec.Source == "" {
+		var newSource string
```

Comment:
nit: can't we just use a `:=` in the next line and not define `newSource` up front?

Reply:
It runs into an issue because `err` is already defined.

Reply:
`err is shadowed during returngo`

Reply:
aye. okay, then not for this pr but we should be stricter about using return shadowing in general... I'm not generally a fan of it for reasons like this.

Reply:
Agreed.